File: ck-editor5-image-upload-master/Client/src/service/CategoryService.js
import axios from 'axios';

const CATEGORY_REST_API ="http://localhost:8080/category";

class CategoryService{

    getCategory(){
        return axios.get(CATEGORY_REST_API);
    }
    createCategory(category){
        return axios.post(CATEGORY_REST_API,category);
    }
    getCategoryById(id){
        return axios.get(CATEGORY_REST_API+'/'+id);
    }
    deleteCategory(id){
        return axios.delete(CATEGORY_REST_API+'/'+id);
    }
    updateCategory(category,id){
        return axios.put(CATEGORY_REST_API+'/'+id,category);
    }
}
export default new CategoryService();

File: daoduytinhjavaweb/src/main/java/com/daoduytinh/service/CategoryService.java
package com.daoduytinh.service;

import java.util.List;
import java.util.Map;

import javax.validation.Valid;

import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;

import com.daoduytinh.entity.Category;
import com.daoduytinh.exception.ResourceNotFoundException;

@Service
public interface CategoryService {
	
	public List<Category> getAllCategory();
	
	public ResponseEntity<Category> getCategoryById(@PathVariable(value = "id") Long Id)
	        throws ResourceNotFoundException;
	
	public Category createCategory(@Valid @RequestBody Category category);
	
	public ResponseEntity<Category> updateCategory(@PathVariable(value = "id") Long Id,
	         @Valid @RequestBody Category categoryDetail) throws ResourceNotFoundException;
	
	public Map<String, Boolean> deleteCategory(@PathVariable(value = "id") Long Id)
	         throws ResourceNotFoundException;
}
